eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/Handler.php

```diff
@@ -1004,4 +1013,66 @@ protected function getHash($text)
     {
         return md5(mb_strtolower($text, 'UTF-8'));
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function archiveUrlAliasesForDeletedTranslations($locationId, $parentLocationId, array $languageCodes)
+    {
+        $parentId = $this->getRealAliasId($parentLocationId);
+
+        // filter removed Translations
+        $urlAliases = $this->listURLAliasesForLocation($locationId);
+        $removedLanguages = [];
+        foreach ($urlAliases as $urlAlias) {
+            $removedLanguages = array_merge(
+                $removedLanguages,
+                array_filter(
+                    $urlAlias->languageCodes,
+                    function ($languageCode) use ($languageCodes) {
+                        return !in_array($languageCode, $languageCodes);
+                    }
+                )
+            );
+        }
```

Comment:
Logic behind lines 1027-1037 could be written as well using nested loops: 

```php
foreach ($urlAliases as $urlAlias) {
    foreach ($urlAlias->languageCodes as $languageCode) {
        if (!in_array($languageCode, $languageCodes)) {
            $removedLanguages[] = $languageCode;
        }
    }
}
```

Am I right?

Reply:
Yeah, that's more readable. Thanks.